File: 1.Fundamentals/Remove Nth Node From End of List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    //silding window O()
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode dummyHead = new ListNode(0);
        dummyHead.next = head;
        ListNode pre = dummyHead;
        ListNode cur = dummyHead;
        
        while(n >= 0){
            cur = cur.next;
            n--;
        }
        
        while(cur != null){
            cur = cur.next;
            pre = pre.next;
        }
        
        pre.next = pre.next.next;
        
        return dummyHead.next;
        
    }
}

File: 1.Fundamentals/Remove Nth Node From End of List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def removeNthFromEnd(self, head: ListNode, n: int) -> ListNode:
        dummyHead = ListNode(0);
        dummyHead.next = head;
        cur = dummyHead;
        pre = dummyHead;
        
        while n >= 0:
            cur = cur.next
            n -= 1
        
        while cur != None:
            cur = cur.next
            pre = pre.next
        
        pre.next = pre.next.next
        
        return dummyHead.next